docs/esp32/tutorial/reset.rst

```diff
@@ -0,0 +1,25 @@
+Factory reset
+=============
+
+If something unexpected happens and your ESP32-based board no longer boots
+MicroPython, then you may have to factory reset it. For more details, see
+:ref:`soft_bricking`.
+
+Factory resetting the MicroPython esp32 port involves fully erasing the flash
+and resetting the flash memory, so you will need to re-flash the MicroPython
+firmware afterwards and copy any Python files to the filesystem again.
+
+1. You will need the Espressif `esptool`_ installed on your system. This is the
+   same tool that you may have used to initially install MicroPython on your
+   board (see :ref:`installation instructions <esp32_flashing>`).
+2. Find the serial port name of your board, and then use esptool to erase the
+   entire flash contents::
+
+       esptool.py -p PORTNAME erase_flash
```

Comment:
For most users these days the auto-detection of port works well - it's often only advanced users that have more than one com port on their computer at any one time. As such I'd remove the need to "find serial port name" as new users wont know how to do this. 
If desired this could be split to show the basic `esptool.py erase_flash` first and then a short follow up paragraph on setting the port name as well if needed?

Reply:
*Tries to imagine a world where there aren't nine serial devices plugged into a computer at all times*
![image](https://github.com/user-attachments/assets/66a4372e-a788-406d-96fd-bbcee2a6593e)

:grin: In all seriousness, I think I agree. However, I suggest I submit a separate PR for this that also changes it in the esp32 tutorial. Also there are the download page instructions, although I'm not sure where these are generated from - maybe damien has to make that change somewhere internally.

Reply:
> Also there are the download page instructions, although I'm not sure where these are generated from

They should be in the various `deploy.md` files (part of the `board.json` and `board.md` set of files).